python/lsst/ap/association/dia_object.py

```diff
@@ -231,6 +262,16 @@ def dia_source_catalog(self):
         return self._dia_source_catalog
 
     @property
+    def n_dia_sources(self):
```

Comment:
Perhaps but I thought it would be useful to be explicit as to which length was being returned. __len__ could be interpreted as returning a property of the the DIAObject not necessarily the number of dia_sources in this object.